### PR TITLE
Max Cap, Gas Fee estimation, Allowance Fix

### PR DESCRIPTION
**Pull request by 0xAplki**
## Description of Changes
- applied Max Cap + button copy
- Refactored those using exhaustive-reps escape for bug resiliency
- Gas fee estimation
- Allowance fix
- Shuttle fee formatting fix
- Resolved conflicts

Sample WIP Implementation
https://drive.google.com/file/d/1czUhzBRAUaaQMHm9A3qJZqADl9mlSB3J/view

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-225)

### Developer Checklist:

* [x] I have followed the guidelines in our Contributing document
* [x] This PR has a corresponding JIRA ticket
* [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
* [x] All files have appropriate file headers and documentation
* [x] I have written new tests for your core changes, as applicable
* [x] I have successfully ran tests locally

### Reviewers Checklist:
* [ ] Code is readable and understandable; any unclear parts have explanations 
* [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
* [ ] I have successfully ran tests locally

### CODE REVIEW

**Comment by mystbrent** [comment]
Just noticed there are conflicts with develop

**Comment by 0xAplki** [comment]
Updated:
- Shuttle fee formatting fix
- Resolved conflicts


Ready for review!

**Comment by 0xAplki** [comment]
please do hard refresh before testing :)

**Review by tracyarciaga — COMMENTED** [review]
@0xAplki Tried to test in preview link and in local too. I got this issue -> Bridge crashes after selecting an asset and switching network: 
CleanShot 2021-09-09 at 09 01 44

**Review by bitcoinbrisbane — APPROVED** [review]
LGTM

File: src/halo-hooks/useBridge.ts
Comment (schystz) [inline src/halo-hooks/useBridge.ts]: `chainId as ChainId` is no longer necessary right? The type should already be `ChainId`, it was previously imported using `useWeb3React()` hence the need for type casting.

Noticed that this is repeatedly used throughout this file.

File: src/halo-hooks/useBridge.ts
Comment (schystz) [inline src/halo-hooks/useBridge.ts]: Won't this cause multiple loops? If not please disregard.

File: src/halo-hooks/useBridge.ts
Comment (schystz) [inline src/halo-hooks/useBridge.ts]: I think we don't need to log this event, since we already log deposit and burn. We don't log this event on Farm or Liquidity Pools page currently.

File: src/pages/Tailwind/Bridge/BridgePanel.tsx
Comment (schystz) [inline src/pages/Tailwind/Bridge/BridgePanel.tsx]: Shall we use a more user-friendly message? "Input" can be misunderstood as a verb instead of a noun. I think something simple like "Maximum amount reached" is easier to understand.

File: src/pages/Tailwind/Bridge/BridgePanel.tsx
Comment (schystz) [inline src/pages/Tailwind/Bridge/BridgePanel.tsx:248]: Can we use else-if here, and on the series of "if"s statement above?

File: src/utils/bridge.ts
Comment (schystz) [inline src/utils/bridge.ts]: Shall we have a fallback value here just in case `REACT_APP_SHUTTLE_FEE_PERCENTAGE` is not set? Something like: `process.env.REACT_APP_SHUTTLE_FEE_PERCENTAGE ?? 0`

**Review by schystz — COMMENTED** [review]
Looks good overall, few minor comments

File: src/halo-hooks/useBridge.ts
Comment (0xAplki) [reply src/halo-hooks/useBridge.ts]: gotcha

File: src/pages/Tailwind/Bridge/BridgePanel.tsx
Comment (0xAplki) [reply src/pages/Tailwind/Bridge/BridgePanel.tsx]: This looks good will update!

File: src/pages/Tailwind/Bridge/BridgePanel.tsx
Comment (0xAplki) [reply src/pages/Tailwind/Bridge/BridgePanel.tsx:248]: this is updated. was just fixing some merge conflicts as well

**Review by schystz — APPROVED** [review]
Looks good now!

**Review by tracyarciaga — APPROVED** [review]
Approving this let's fix the bug after merge

**Pull request merged**
